packages/url_launcher/url_launcher_windows/test/url_launcher_windows_test.dart

```diff
@@ -5,140 +5,99 @@
-        log,
-        <Matcher>[
-          isMethodCall('launch', arguments: <String, Object>{
-            'url': 'http://example.com/',
-            'enableJavaScript': false,
-            'enableDomStorage': false,
-            'universalLinksOnly': false,
-            'headers': <String, String>{'key': 'value'},
-          })
-        ],
-      );
+      final bool result = await plugin.canLaunch('http://example.com/');
+
+      expect(result, isFalse);
+      expect(api.argument, 'http://example.com/');
     });
+  });
+
+  group('launch', () {
+    test('handles success', () async {
+      api.canLaunch = true;
 
-    test('launch universal links only', () async {
-      final UrlLauncherWindows launcher = UrlLauncherWindows();
-      await launcher.launch(
-        'http://example.com/',
-        useSafariVC: false,
-        useWebView: false,
-        enableJavaScript: false,
-        enableDomStorage: false,
-        universalLinksOnly: true,
-        headers: const <String, String>{},
-      );
       expect(
-        log,
-        <Matcher>[
-          isMethodCall('launch', arguments: <String, Object>{
-            'url': 'http://example.com/',
-            'enableJavaScript': false,
-            'enableDomStorage': false,
-            'universalLinksOnly': true,
-            'headers': <String, String>{},
-          })
-        ],
-      );
+          plugin.launch(
+            'http://example.com/',
+            useSafariVC: true,
+            useWebView: false,
+            enableJavaScript: false,
+            enableDomStorage: false,
+            universalLinksOnly: false,
+            headers: const <String, String>{},
+          ),
+          completes);
+      expect(api.argument, 'http://example.com/');
     });
 
-    test('launch should return false if platform returns null', () async {
-      final UrlLauncherWindows launcher = UrlLauncherWindows();
-      final bool launched = await launcher.launch(
-        'http://example.com/',
-        useSafariVC: true,
-        useWebView: false,
-        enableJavaScript: false,
-        enableDomStorage: false,
-        universalLinksOnly: false,
-        headers: const <String, String>{},
-      );
-
-      expect(launched, false);
+    test('handles failure', () async {
+      api.canLaunch = false;
+
+      await expectLater(
+          plugin.launch(
+            'http://example.com/',
+            useSafariVC: true,
+            useWebView: false,
+            enableJavaScript: false,
+            enableDomStorage: false,
+            universalLinksOnly: false,
+            headers: const <String, String>{},
+          ),
+          throwsA(isA<PlatformException>()));
+      expect(api.argument, 'http://example.com/');
     });
   });
 }
+
+class _FakeUrlLauncherApi implements UrlLauncherApi {
+  /// The argument that was passed to an API call.
+  String? argument;
+
+  /// The return value for [canLaunchUrl], and whether or not [launchUrl]
+  /// should throw.
```

Comment:
Maybe something like this?
```dart
/// The return value for [canLaunchUrl]. If `false`, [launchUrl] throws.
```

Reply:
I rewrote it entirely based on your suggestion, but expanded more to make it less confusing:

```
  /// Controls the behavior of the fake implementations.
  ///
  /// - [canLaunchUrl] returns this value.
  /// - [launchUrl] throws if this is false.
```